File: server/event_schema.py
from pydantic import BaseModel
from typing import Optional, Dict, Any, Literal


class WorkflowEvent(BaseModel):
    type: Literal['agent_token', 'agent_message', 'step_update', 'error', 'done']
    conversation_id: str
    workflow_run_id: str
    step_id: str
    agent_id: str
    delta: Optional[str] = None
    content: Optional[str] = None
    state_patch: Optional[Dict[str, Any]] = None
    metadata: Optional[Dict[str, Any]] = None


__all__ = ['WorkflowEvent']

File: client/src/lib/eventSchema.ts
export type WorkflowEvent = {
  type: 'agent_token' | 'agent_message' | 'step_update' | 'error' | 'done'
  conversation_id: string
  workflow_run_id: string
  step_id: string
  agent_id: string
  delta?: string
  content?: string
  state_patch?: Record<string, unknown>
  metadata?: Record<string, unknown>
}
